### PR TITLE
feat(core-flows,medusa,types): remove rules from promotion endpoints + workflows

### PR DESCRIPTION
**Pull request by riqwan**
what:

- adds endpoints to remove rules from a promotion:
  - promotion rules
  - target rules
  - buy rules

- updates promotion remove*Rules method to simply accept rule_ids to delete instead of array of objects with just id

### CODE REVIEW

File: integration-tests/modules/__tests__/promotion/admin/promotion-rules.spec.ts [hunk 1]
```diff
@@ -342,6 +342,194 @@ medusaIntegrationTestRunner({
           )
```
Comment (olivermrbl) [inline integration-tests/modules/__tests__/promotion/admin/promotion-rules.spec.ts:342]: comment(this file): At some point, this file should be moved to `integration-tests/api`

**Review by olivermrbl — APPROVED** [review]
Nice work @riqwan

File: integration-tests/modules/__tests__/promotion/admin/promotion-rules.spec.ts (hunk 1 above)
Comment (riqwan) [reply integration-tests/modules/__tests__/promotion/admin/promotion-rules.spec.ts:342]: Shouldn't we wait for the CI to run v2 api before doing that?

Comment (olivermrbl) [reply integration-tests/modules/__tests__/promotion/admin/promotion-rules.spec.ts:342]: Yeah, that's true

**Pull request merged**